app/assets/javascripts/flux/image_store.js.jsx

```diff
@@ -49,6 +58,53 @@ class ImageStore {
       this._listeners.splice(index, 1);
     }
   }
+
+  upvote(id) {
+    console.log(id);
+    var my_image = _.find(this._images, function(obj) { return obj.id == id });
+    my_image.upvoted = true;
+    API.postUpvote(
+      id,
+      (success) => {
+        console.log("success");
+      },
+      (error) => {
+        console.error('Error upvote');
+      }
+    );
+      if (this.isDownvoted(id)) {
+        my_image.score++;
```

Comment:
Shouldn't you set `my_image.downvoted` to False here? Or even better, use a ternary value (`status=oneOf("UP", "DOWN", "NONE")`) to show that these states are exclusive.